### DISCUSSION TITLE
Ability to disable some beans in the Spring Boot auto configuration?

### DISCUSSION THREAD

**Discussion by aukevanleeuwen** [zalando/logbook#1840] [Ideas]
Looking at the Spring Boot autoconfiguration it's not all that easy to remove some of the default filters. As an example: 

`bodyFilters` has an explicit property to disable:
https://github.com/zalando/logbook/blob/fde0074fee77fd2b757df58d894f09d8317494ff/logbook-spring-boot-autoconfigure/src/main/java/org/zalando/logbook/autoconfigure/LogbookAutoConfiguration.java#L249-L256

and `pathFilter` defaults to `none()`:
https://github.com/zalando/logbook/blob/fde0074fee77fd2b757df58d894f09d8317494ff/logbook-spring-boot-autoconfigure/src/main/java/org/zalando/logbook/autoconfigure/LogbookAutoConfiguration.java#L236-L248

but the potentially expensive JSON parsing cannot be easily disabled, because it's conditional on a `JacksonJsonFieldBodyFilter`:
https://github.com/zalando/logbook/blob/fde0074fee77fd2b757df58d894f09d8317494ff/logbook-spring-boot-autoconfigure/src/main/java/org/zalando/logbook/autoconfigure/LogbookAutoConfiguration.java#L257-L266

You can specify a bean with exactly the same name `jsonBodyFieldsFilter`, but this requires you to set `spring.main.allow-bean-definition-overriding=true`, which ideally I don't want to do and is quite brittle as well. Or do something like:

```java
    @Bean
    public JacksonJsonFieldBodyFilter filter() {
        return new JacksonJsonFieldBodyFilter(List.of(), "") {
            @Override
            public String filter(@Nullable String contentType, String body) {
                return body;
            }
        };
    }
```

which is also not very nice. Maybe adding properties to them like the body filter has makes sense?

**Comment by whiskeysierra** [comment]
The `JacksonJsonFieldBodyFilter` depends on `obfuscate.getJsonBodyFields()` which defaults to an empty list.
I like how the `PathFilter` has the same thing with `paths` but checks if they are empty and defaults to a *no-op*.
The JSON filter should do the same imho and fallback to `BodyFilter.none()`.

**Reply by aukevanleeuwen** [reply]
That makes sense I suppose.

**Reply by aukevanleeuwen** [reply]
For future readers: I think this is now possible thanks to @skjolber : https://github.com/zalando/logbook/pull/2007
